### PR TITLE
docs(checkbox): implement and document vars and attrs

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

`hx-checkbox` : implement and document variables and attributes

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2102

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [ ] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/scss/components/checkbox/_index.scss [hunk 1]
```diff
@@ -120,11 +120,11 @@ hx-checkbox-control {
 
     &:checked,
     &:indeterminate {
-      color: $cyan-900;
+      color: var(--hxCheckbox-indeterminate, $cyan-900);
```
Comment (lalithkarikelli) [inline src/scss/components/checkbox/_index.scss:123]: please change to var(--hxCheckbox-indeterminate-color, $cyan-900);

File: src/scss/components/checkbox/_index.scss [hunk 2]
```diff
@@ -120,11 +120,11 @@ hx-checkbox-control {
...
 
       ~ label > hx-checkbox {
-        border-color: $cyan-900;
-        color: $cyan-900;
+        border-color: var(--hxCheckbox-indeterminate-label-bordercolor, $cyan-900);
+        color: var(--hxCheckbox-indeterminate-label, $cyan-900);
```
Comment (lalithkarikelli) [inline src/scss/components/checkbox/_index.scss:127]: please change to var(--hxCheckbox-indeterminate-label-color, $cyan-900);

File: src/scss/components/checkbox/_index.scss [hunk 3]
```diff
@@ -179,7 +179,7 @@ hx-checkbox-control > input[type="checkbox"]:disabled {
   }
 
   ~ label {
-    color: $gray-700;
+    color: var(--hxCheckbox-disabled-label, $gray-700);
```
Comment (lalithkarikelli) [inline src/scss/components/checkbox/_index.scss:182]: please change to var(--hxCheckbox-disabled-label-color, $gray-700);

File: src/scss/components/checkbox/mixins/_hxCheckboxControl.scss [hunk 4]
```diff
@@ -8,22 +8,22 @@
     box-shadow: $focus-glow;
 
     &::-ms-check {
-      border-color: $cyan-700;
+      border-color: var(--hxCheckbox-bordercolor-focus, $cyan-700);
```
Comment (lalithkarikelli) [inline src/scss/components/checkbox/mixins/_hxCheckboxControl.scss:11]: please follow same pattern through out the application .

change it to var(--hxCheckbox-focus-bordercolor, $cyan-700);

**Review by lalithkarikelli — CHANGES_REQUESTED** [review]

File: src/scss/components/checkbox/mixins/_hxCheckboxControl.scss (hunk 4 above)
Comment (BY00565233) [reply src/scss/components/checkbox/mixins/_hxCheckboxControl.scss]: Changes have been done, Please review

File: src/scss/components/checkbox/_index.scss (hunk 3 above)
Comment (BY00565233) [reply src/scss/components/checkbox/_index.scss]: Changes have been done, Please review

File: src/scss/components/checkbox/_index.scss (hunk 2 above)
Comment (BY00565233) [reply src/scss/components/checkbox/_index.scss]: Changes have been done, Please review

File: src/scss/components/checkbox/_index.scss (hunk 1 above)
Comment (BY00565233) [reply src/scss/components/checkbox/_index.scss]: Changes have been done, Please review

**Review by 100stacks — CHANGES_REQUESTED** [review]
@BY00565233, please rebase and fix merge conflicts.

**Review by 100stacks — CHANGES_REQUESTED** [review]
### Please rebase.

**Review by lalithkarikelli — APPROVED** [review]
DEV LGTM

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**